Mathlib/CategoryTheory/Category/Preorder.lean

```diff
@@ -0,0 +1,201 @@
+
+/-- Construct a morphism in the opposite of a preorder category from an inequality. -/
+def opHomOfLe {x y : Xᵒᵖ} (h : unop x ≤ unop y) : y ⟶ x :=
+  (homOfLE h).op
+#align category_theory.op_hom_of_le CategoryTheory.opHomOfLe
+
+theorem le_ofOp_hom {x y : Xᵒᵖ} (h : x ⟶ y) : unop y ≤ unop x :=
+  h.unop.le
+#align category_theory.le_of_op_hom CategoryTheory.le_ofOp_hom
+
+instance uniqueToTop [OrderTop X] {x : X} : Unique (x ⟶ ⊤) where 
+  default := homOfLE le_top 
+  uniq := fun a => by rfl  
+#align category_theory.unique_to_top CategoryTheory.uniqueToTop
+
+instance uniqueFromBot [OrderBot X] {x : X} : Unique (⊥ ⟶ x) where 
+  default := homOfLE bot_le 
+  uniq := fun a => by rfl 
+#align category_theory.unique_from_bot CategoryTheory.uniqueFromBot
+
+end CategoryTheory
+
+section
+
+variable {X : Type u} {Y : Type v} [Preorder X] [Preorder Y]
+
+/-- A monotone function between preorders induces a functor between the associated categories.
+-/
+def Monotone.functor {f : X → Y} (h : Monotone f) : X ⥤ Y
+    where
+  obj := f
+  map := @fun x₁ x₂ g => CategoryTheory.homOfLE (h g.le)
+#align monotone.functor Monotone.functor
+
+@[simp]
+theorem Monotone.functor_obj {f : X → Y} (h : Monotone f) : h.functor.obj = f :=
+  rfl
+#align monotone.functor_obj Monotone.functor_obj
+
+end
+
+namespace CategoryTheory
+
+section Preorder
+
+variable {X : Type u} {Y : Type v} [Preorder X] [Preorder Y]
+
+/-- A functor between preorder categories is monotone.
+-/
+-- @[mono] porting note: `mono` tactic is not ported yet
+theorem Functor.monotone (f : X ⥤ Y) : Monotone f.obj := fun _ _ hxy => (f.map hxy.hom).le
+#align category_theory.functor.monotone CategoryTheory.Functor.monotone
+
+end Preorder
+
+section PartialOrder
+
+variable {X : Type u} {Y : Type v} [PartialOrder X] [PartialOrder Y]
+
+theorem Iso.to_eq {x y : X} (f : x ≅ y) : x = y :=
+  le_antisymm f.hom.le f.inv.le
+#align category_theory.iso.to_eq CategoryTheory.Iso.to_eq
+
+/-- A categorical equivalence between partial orders is just an order isomorphism.
+-/
+def Equivalence.toOrderIso (e : X ≌ Y) : X ≃o Y
+    where
+  toFun := e.functor.obj
+  invFun := e.inverse.obj
+  left_inv a := (e.unitIso.app a).to_eq.symm
+  right_inv b := (e.counitIso.app b).to_eq
+  map_rel_iff' := @fun a a' =>
+    ⟨fun h =>
+      ((Equivalence.unit e).app a ≫ e.inverse.map h.hom ≫ (Equivalence.unitInv e).app a').le,
+      fun h : a ≤ a' => (e.functor.map h.hom).le⟩
+#align category_theory.equivalence.to_order_iso CategoryTheory.Equivalence.toOrderIso
+
+-- `@[simps]` on `Equivalence.toOrderIso` produces lemmas that fail the `simpNF` linter,
+-- so we provide them by hand:
+@[simp]
+theorem Equivalence.toOrderIso_apply (e : X ≌ Y) (x : X) : e.toOrderIso x = e.functor.obj x :=
+  rfl
+#align category_theory.equivalence.to_order_iso_apply CategoryTheory.Equivalence.toOrderIso_apply
+
+@[simp]
+theorem Equivalence.toOrderIso_symm_apply (e : X ≌ Y) (y : Y) :
+    e.toOrderIso.symm y = e.inverse.obj y :=
+  rfl
```

Comment:
Started a zulip discussion about this issue: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simps.20and.20simpNF.20linter